Initial PR for bazel sidecar

gattytto opened the pull request:
Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>

Initial contribution for bazel sidecar

Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>

gattytto committed:
trying to comply with new structure

Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>

gattytto committed:
trying to comply with new structure

Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>

ericwill commented on Dockerfile:
```diff
@@ -0,0 +1,47 @@
+    chgrp -R 0 /run && chmod -R g+rwX /run && \
+    curl https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl && \
+    chmod +x /usr/local/bin/kubectl && \
+    curl -o- -L https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz | tar xvz -C /usr/local/bin --strip 1 && \
```
You don't need `helm` and `kubectl` in this image, it's part of [che-container-tools](https://github.com/che-dockerfiles/che-container-tools/blob/master/Dockerfile)

ericwill commented on Dockerfile:
```diff
@@ -0,0 +1,47 @@
+
+RUN cd /tmp && wget https://github.com/bazelbuild/buildtools/releases/download/${BUIDLERS_VERSION}/buildifier && chmod 777 buildifier && mv buildifier /usr/bin/
+
+RUN cd /tmp && wget https://github.com/bazelbuild/buildtools/releases/download/${BUIDLERS_VERSION}/buildozer && chmod 777 buildozer && mv buildozer /usr/bin/
```
Let's merge these 3 run steps into the first one (one run step)

gattytto committed:
removing unnecessary stuff

Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>
Removing stuff already present in the extended base image as @ericwill pointed out 
Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>

gattytto committed:
more cleaning

Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>
let's see if it builds now
Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>

gattytto committed:
Update Dockerfile

ericwill commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+    cd /tmp && wget https://github.com/bazelbuild/buildtools/releases/download/${BUIDLERS_VERSION}/buildifier && chmod 777 buildifier && mv buildifier /usr/bin/ && \
+    cd /tmp && wget https://github.com/bazelbuild/buildtools/releases/download/${BUIDLERS_VERSION}/buildozer && chmod 777 buildozer && mv buildozer /usr/bin/
+
+ADD etc/entrypoint.sh /entrypoint.sh
```
This statement and the ones below it are unneeded, we have it in the base image already: https://github.com/che-dockerfiles/che-container-tools/blob/master/Dockerfile#L51

gattytto committed:
Remove repeated statements

Remove repeated statements

Solves https://github.com/che-dockerfiles/che-sidecar-bazel/pull/4#discussion_r461555103

Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>

ericwill commented on etc/entrypoint.sh:
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
```
You can delete this file, nothing uses it in this repo

ericwill commented:
> Closes [eclipse/che#15712](https://github.com/eclipse/che/issues/15712)

I would change this, once this PR is merged there is still the PR for the plugin registry

gattytto committed:
delete unnecessary entrypoint

Signed-off-by: Esteban Mañaricua <emanaricua@gmail.com>
